src/main/java/org/openstreetmap/atlas/checks/validation/areas/TallBuildingCheck.java

```diff
@@ -0,0 +1,728 @@
+        {
+            final Optional<Double> lowerQuartile = this.getLowerQuartile(sortedTagValueList);
+            final Optional<Double> upperQuartile = this.getUpperQuartile(sortedTagValueList);
+            final Optional<Double> innerQuartileRange = this
+                    .getInnerQuartileRange(sortedTagValueList);
+
+            if (lowerQuartile.isPresent() && upperQuartile.isPresent()
+                    && innerQuartileRange.isPresent())
+            {
+                this.storedAreasWithStatistics.put(bufferArea, new Tuple4<>(tagIdentifier,
+                        lowerQuartile.get(), upperQuartile.get(), innerQuartileRange.get()));
+                if (this.isOutlier(tagValue, lowerQuartile.get(), upperQuartile.get(),
+                        innerQuartileRange.get()) && tagIdentifier.equals(BuildingLevelsTag.KEY))
+                {
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(2, object.getOsmIdentifier())));
+                }
+
+                if (this.isOutlier(tagValue, lowerQuartile.get(), upperQuartile.get(),
+                        innerQuartileRange.get()) && tagIdentifier.equals(HeightTag.KEY))
+                {
+                    return Optional.of(this.createFlag(object, this.getLocalizedInstruction(
+                            (int) this.magicNumber3, object.getOsmIdentifier())));
+                }
+
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Function to get third quartile value from sorted list of relevant tags.
+     * 
+     * @param listOfRelevantTags
+     *            sorted list of relevant tags
+     * @return third quartile value of dataset (tags)
+     */
+    private Optional<Double> getUpperQuartile(final List<Double> listOfRelevantTags)
+    {
+        final int lengthOfList = listOfRelevantTags.size() - 1;
+        if (lengthOfList == 0)
+        {
+            return Optional.empty();
+        }
+        return Optional.of(listOfRelevantTags
+                .get((int) Math.round(lengthOfList * this.magicNumberThreeQuarters)));
+    }
+
+    /**
+     * Function to determine if building has "building:levels" tag.
+     * 
+     * @param tags
+     *            building tags
+     * @return boolean if it has specific tag.
+     */
+    private boolean hasBuildingLevelTag(final Map<String, String> tags)
+    {
+        return tags.containsKey(BuildingLevelsTag.KEY);
+    }
+
+    /**
+     * Function to determine if building has "height" tag.
+     * 
+     * @param tags
+     *            building tags
+     * @return boolean if it has specific tag.
+     */
+    private boolean hasHeightTag(final Map<String, String> tags)
+    {
+        return tags.containsKey(HeightTag.KEY);
+    }
+
+    private boolean hasInvalidHeightTag(final String heightTag)
+    {
+        return (!heightTag.contains(" ") && heightTag.contains("m"))
```

Comment:
What is the reasoning for `(!heightTag.contains(" ") && heightTag.contains("m"))`

Reply:
this is a specific height tag case that needs to be modeled with with a numerical value, then a space, then a unit of measurement according to [Osm Wiki](https://wiki.openstreetmap.org/wiki/Key:height) for height tags. I am just capturing incorrectly modeled tags such as -> "5m". That tag should be -> "5 m".

Reply:
@reichg Got it! Is there possibility of other unit of measurement usage for height tag for different countries?

Reply:
Technically yes but meters is the default and takes up about 95% of the values that belong to the key.